intents/intent.go

```diff
@@ -12,6 +16,14 @@ import (
 	"github.com/0xsequence/ethkit/go-ethereum/crypto"
 )
 
+var (
+	ErrInvalidPacket    = fmt.Errorf("invalid packet")
+	ErrNoSignatures     = fmt.Errorf("no signatures")
+	ErrInvalidSignature = fmt.Errorf("invalid signature")
+)
+
+type IntentVerifierGetter func(sessionId string) (string, error)
```

Comment:
btw, I still cant find the code in the PRs how the sessionID and sessionVerifier are related...?

The sessionID is a UUID, and the sessionVerifier is what..? the public key or address..?

I would propose, we just stick to the name `sessionAddress` and just add `sessionAlg` or `alg` for short with values `P256K1` or `P256R1` for the alg, and for the `sessionAddress` we have the hex value of either the K1 wallet address (42 characters which includes the 0x prefix), or the R1 public key (132 characters which includes the 0x prefix).

Lmk what you think..?

Some other ideas:
* We could bake the "alg" into the session address, and give session address a different name
* For example, call it `sessionPubkey` and reserve the first byte for the alg version, so we'd have... `0x`+algByte+pubkeyOrAddress
* So, say we have a session address of 0x95222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5 .. then our `sessionPubkey` would be '0x'+'01'+'95222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5' where 01 is considered k1
* For r1 we'd use alg '02' so .. sessionPubkey would look like: '0x'+'02'+pubKeyForR1KeyHere

lmk what you prefer..? I kinda like baking it together as it'll be viewer arguments to update, and we just rename `session_address` to `session_pubkey` and we're done. We also know, that k1 alg of 01 means its followed by the address... if for any reason we want to adapt in the future we can have additional 03, 04, etc.. spaces

Reply:
as well... what if we base64 (url-safe variant) encoded the session_id value... for the r1 alg..?

in which case, it would get a whole lot shorter... 65 bytes down to around ~28 string character.. 

if we decided to do this... then some ideas:

* lets drop the '0x' prefix, as the value won't always be hex, it will just be a string
* Lets keep the alg byte prefix, so '01' and '02' values
* For k1 curve, above session_id would be ....... `0195222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5`
* For r1 curve, above session_id would be ... `02`+base64UrlSafeVariant(pubKeyRaw) .. which as example would look something like `02W29iamVjdCBBcnJheUJ1ZmZlcl0` -- and FYI, for base64 url safe variant in go and js, see https://github.com/goware/base64

Reply:
The ''IntentVerifierGetter'' is required to get the ''publicKey'' to verify the signature. As an input it accepts the ``sessionId`` and outputs ``sessionVerifier`` which in P256K1 is ``sessionAddress`` and in P256R1 is ``publicKey``. As I mentioned in replies in other PR I don't like the idea of using ``address`` in here. I will be going towards ``sessionPublicKey``. It will be much clearer.

If it comes to this verification logic. I want to remodel it because I do not like how it looks. I do not like this getter here.

The idea to store ``keyType`` as a first byte of the pub key sounds good. I have used prefix before but byte works too.

I have been experimenting with base64 encoding for pub key, however it does not make much sense if this pub key is sent only once on ``openSession``. Later ``sessionId`` is always used to reference the session and BE can look up the pub key required for verification.

NOTE: Using ``sessionId`` and ``sessionPublicKey`` instead of just ``sessionAddress`` is a separation of concerns. The ``sessionId`` is a short consistent session identifier and ``sessionPublicKey`` is a means for session signature verification. We already mixed up those two and made a mess in the API where we have been using ``sessionId`` / ``sessionAddress`` names interchangeably. Making it difficult to store(fixed length, requirement to start with 0x), and inconsistent in size and form.

Reply:
@patrislav has convinced me it will be faster, and easier to implement a version where ``sessionId`` = ``publicKey``. So this is what I'm going to do.

Reply:
I propose:
1. keep only `sessionId` -- we do not need both address and id. We should deprecate the other.
2. `sessionId` should represent the public key, most definitely. I think a alg prefix is the right idea, because it allows multi-curve support (like ipfs folks do, and it works well)
3. base64 url-variant encoding either of the public key for k1 and r1, (or leaving k1 address inside of the sessionId), is helpful because it shortens the string. The only benefit here is we do specify the `sessionId` inside of the oauth idtoken, so it will make that dataset longer..? however... maybe it would be fine since the idtoken is base64 url variant encoded itself..?

I feel (1) and (2) are absolutely the right thing to do.. but I'm curious what you guys think about (3)

Reply:
At the moment I'm doing 1. & 2. + integration of account system and authenticator rpc-v1. Once all of this is done I will have a look at base64url encoding. 

If we are doing 1. & 2. then it makes sense to use base64url as it will make each call body significantly smaller. Other than that it does not matter that much.